src/web/blocks.rs

```diff
@@ -0,0 +1,160 @@
+use crate::iroha_client_wrap::QueryBuilder;
+
+use super::{
+    etc::{HashDeser, SerScaleHex, Timestamp},
+    get,
+    pagination::{Paginated, PaginationQueryParams},
+    web, AppData, Scope, WebError,
+};
+use color_eyre::{
+    eyre::{eyre, Context},
+    Result,
+};
+use iroha_core::{
+    prelude::VersionedValidTransaction,
+    tx::{Pagination, VersionedRejectedTransaction},
+};
+use iroha_crypto::Hash;
+use iroha_data_model::prelude::{BlockValue, FindAllBlocks};
+use serde::Serialize;
+use std::num::NonZeroU64;
+
+/// Block DTO intended to be lightweight and to have only simple aggregated data.
+/// Detailed data is contained within [`BlockDTO`]
+#[derive(Serialize)]
+pub struct BlockShallowDTO {
+    /// See [`BlockDTO`]'s height
+    height: u32,
+    timestamp: Timestamp,
+    block_hash: SerScaleHex<Hash>,
+    transactions: u32,
+    rejected_transactions: u32,
+}
+
+impl TryFrom<BlockValue> for BlockShallowDTO {
+    type Error = color_eyre::Report;
+
+    fn try_from(block: BlockValue) -> Result<Self> {
+        Ok(Self {
+            height: block.header.height.try_into()?,
+            block_hash: block.header.current_block_hash.into(),
+            timestamp: Timestamp::try_from(block.header.timestamp)?,
+            transactions: block.transactions.len().try_into()?,
+            rejected_transactions: block.rejected_transactions.len().try_into()?,
+        })
+    }
+}
+
+/// Full Block DTO
+#[derive(Serialize)]
+pub struct BlockDTO {
+    /// Original height value is u64, but u64 can't fit into JS `number`
```

Comment:
```suggestion
    // Original height value is u64, but u64 can't fit into JS `number`
```
This is a comment, rather than a doc.